File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int left = 0, right = nums.size() - 1; //initialising two pointers for tracking indexes
        vector<int> res(nums.size());
        for (int i = 0, j = nums.size() - 1; j >= 0; i++, j--) {
            if (nums[i] < pivot) {
                res[left++] = nums[i];
            }
            if (nums[j] > pivot) {
                res[right--] = nums[j];
            }
        }
        while (left <= right) {
            res[left++] = pivot;
        }
        return res;
    }
};

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        //Brute-force approach s.c : O(n), t.c : O(n)
        int[] res = new int[nums.length]; int k = 0;
        for (int x : nums) 
            if (x < pivot)
                res[k++] = x;
        for (int x : nums) 
            if (x == pivot)
                res[k++] = x;
        for (int x : nums) 
            if (x > pivot)
                res[k++] = x;
        for (int i = 0; i < res.length; i++) {
            nums[i] = res[i];
        }
        return nums;
    }
}